core/base/src/TApplication.cxx

```diff
@@ -985,8 +985,9 @@ TString TApplication::GetSetup()
    TString setup = "";
    for (auto& line : lines) {
       setup.Append(line);
-      setup.Append("\n");
+      setup.Append('\n');
    }
+   setup.Chop(); // trim final `\n`
```

Comment:
On the long run, it's maybe better to define a join function? Like https://stackoverflow.com/a/71983949

Reply:
that's a feature request for TString I guess